src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double LIST_EXPAND = 0.5;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList(int number) {
+        if (number <= 0) {
+            throw new ArrayListIndexOutOfBoundsException("can't be 0 or less");
+        }
+        elementData = new Object[number];
+    }
+
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        grow();
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Object remove = elementData[index];
+        System.arraycopy(elementData, index + 1, elementData, index,size - index - 1);
+        size--;
+        return (T) remove;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (element == elementData[i] || element != null && element.equals(elementData[i])) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("Element " + element + " not found");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public void grow() {
```

Comment:
"grow" name means arrays grows every time this method is called. But it checks size before it grows. 
Better name for this method is: "checkSizeAndGrow()".
You can also move "if (elementData.length == size)" out from this method. In this case you can name this method "grow"
```suggestion
    public void checkSizeAndGrow() {
```